graph/denota/build/schema.graphql

```diff
@@ -0,0 +1,237 @@
+# TODO look into ID vs Bytes type for id field
+############################################# Nota ###############################################
+type ERC20 @entity {
+  id: ID!
+  isWhitelisted: Boolean
+  # name: String
+	# symbol: String
+	# decimals: Int!
+  # totalSupply: BigInt!
+}
+
+type Transaction @entity {
+  id: ID!
+  timestamp: BigInt!
+  blockNumber: BigInt!
+  transactionHash: String!
+  events: [Event!]! @derivedFrom(field: "transaction")
+}
+type Account @entity {
+  id: ID!
+  notasSent: [Nota!]! @derivedFrom(field: "sender")
+  notasReceived: [Nota!]! @derivedFrom(field: "receiver")
+  notasOwned: [Nota!]! @derivedFrom(field: "owner")
+
+  transfersFrom: [Transfer!]! @derivedFrom(field: "from")
+  transfersTo: [Transfer!]! @derivedFrom(field: "to")
+
+  approvalsOwner: [Approval!]! @derivedFrom(field: "owner")
+  approvalsApproved: [Approval!]! @derivedFrom(field: "approved")
+  approvalsForAllOwner: [ApprovalForAll!]! @derivedFrom(field: "owner")
+  approvalsForAllOperator: [ApprovalForAll!]! @derivedFrom(field: "operator")
+	events: [Event!]! @derivedFrom(field: "caller")
+  # TODO: make module specific properties addable by module? 
+  # notasInspected: [Nota!]! @derivedFrom(field: "inspector")
+}
+
+interface Event {
+  id: ID!
+  caller: Account!
+  transaction: Transaction!
+}
+
+type Transfer implements Event @entity(immutable: true) {
+  id: ID!
+  caller: Account! # Can be approved/operator..?
+  from: Account!
+  to: Account!
+  nota: Nota!
+  transaction: Transaction!
+}
+
+type Approval implements Event @entity {
+  id: ID!
+  caller: Account!
+  owner: Account!
+  approved: Account!
+  nota: Nota!
+  transaction: Transaction!
+}
+
+type Nota @entity {
+  id: ID!
+  erc20: ERC20
+  escrowed: BigInt
+  module: Module
+  sender: Account
+  receiver: Account
+  owner: Account
+  uri: String
+  # TODO add imageURI: String AND externalURI: String
+  createdTransaction: Transaction
+
+  transfers: [Transfer!]! @derivedFrom(field: "nota")
+  # escrows: [Escrow!]! @derivedFrom(field: "nota")
+  funds: [Funded!]! @derivedFrom(field: "nota")
+  cashes: [Cashed!]! @derivedFrom(field: "nota")
+  approvals: [Approval!]! @derivedFrom(field: "nota")
```

Comment:
can also add `writtens` here

Reply:
There should only be 1 written event possible per Nota which is the one that sets it's state. So prob not needed to track on that entity